app/main.py

```diff
@@ -1,13 +1,24 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        key: value
+        for key, value in {"line": error["line_number"],
+                           "column": error["column_number"],
+                           "message": error["text"],
+                           "name": error["code"],
+                           "source": "flake8"}.items()}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        key: value
+        for key, value in {"errors": [format_linter_error(error)
+                                      for error in errors],
+                           "path": file_path,
+                           "status": "failed" if errors else "passed"}.items()}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(key, error)
```

Comment:
use more informative names for variables
f.e. file_path, errors

Reply:
I'll change *key* to *file_path*, but isn't it better to leave *error* instead of *errors* because it's a for cycle an we go through every item of that dict.